client/client.go

```diff
@@ -382,23 +395,14 @@ func (d *dynamicWatcher) AddOrUpdateWatcher(watcher ObjectIdentifier, watchedObj
 			d.watchedToWatchers[watchedObject] = map[ObjectIdentifier]bool{}
 		}
 
-		// If the object is also watched another object, then do nothing.
+		// If the object is also watched by another object, then do nothing.
 		if _, ok := d.watches[watchedObject]; ok {
 			watchedObjectsSet[watchedObject] = true
 			d.watchedToWatchers[watchedObject][watcher] = true
 
 			continue
 		}
 
-		gvr, err := d.gvrFromObjectIdentifier(watchedObject)
-		if err != nil {
-			klog.Errorf("Could not get the GVR for %s, error: %v", watchedObject, err)
-
-			encounteredErr = err
-
-			break
-		}
-
```

Comment:
I was trying to figure out whether we could keep the code here rather than looking up the watchedObject GVR every single time. Consistency would be good, but what would the consequences be if it stayed here? I'm wondering if there would be a performance improvement not looking up the GVR every time.

Reply:
/unhold I suppose we are caching values, so the performance impact is probably minimal...

Reply:
Yep! The caching is why I felt confident moving it here.

Without moving it, there was the potential for duplicating watches on cluster-scoped resources, if the user supplied different namespaces. I didn't magically see that on my own - it was an observable side effect of the test changes I made and the tests we already had.